test/integration/suite_test.go

```diff
@@ -309,3 +313,47 @@ func waitForBpfdConfigDelete(ctx context.Context, env environments.Environment)
 		}
 	}
 }
+
+func waitForDataplaneReadiness(ctx context.Context, env environments.Environment) error {
+	for {
+		time.Sleep(2 * time.Second)
+		select {
+		case <-ctx.Done():
+			if err := ctx.Err(); err != nil {
+				return fmt.Errorf("context completed while waiting for dataplane ready: %w", err)
+			}
+			return fmt.Errorf("context completed while waiting for dataplane ready")
+		default:
+			fmt.Println("INFO: waiting for dataplane ready")
+
+			dataplanes, err := env.Cluster().Client().CoreV1().Pods(vars.DefaultNamespace).
+				List(ctx, metav1.ListOptions{LabelSelector: fmt.Sprintf("component=%s", vars.DefaultDataPlaneComponentLabel)})
+			if err != nil {
+				return fmt.Errorf("failed to fetch dataplane pod list: %w", err)
+			}
+
+			// Check whether dataplane is set readiness probe.
+			for _, container := range dataplanes.Items[0].Spec.Containers {
```

Comment:
While we would really hope that listing the dataplanes at this point would not return an empty list, it technically could and then we would get an `index out of range` panic here. We throw an error if the listing fails, we could also throw an error if the list succeeds but is unexpectedly empty:

```suggestion
			if len(dataplanes.Items) == 0 {
				return fmt.Errorf("no dataplanes found in namespace %s", vars.DefaultNamespace)
			}

			// Check whether dataplane is set readiness probe.
			for _, container := range dataplanes.Items[0].Spec.Containers {
```